packages/components/src/tree-grid/index.js

```diff
@@ -78,6 +78,10 @@ function TreeGrid(
 			const activeRow = activeElement.closest( '[role="row"]' );
 			const focusablesInRow = getRowFocusables( activeRow );
 			const currentColumnIndex = focusablesInRow.indexOf( activeElement );
+			const canExpandCollapse = 0 === currentColumnIndex;
+			const cannotFocusNextColumn =
+				activeRow.getAttribute( 'aria-expanded' ) === 'false' &&
+				keyCode !== LEFT;
```

Comment:
It seems to work correctly now, and the code is much more readable. Thanks for that!

`cannotFocusNextColumn` seems like it could use `keyCode === RIGHT`. And I think it should also include the `canExpandCollapse` variable in the logic, so that moving focus right is only prevented when a cell in the first column if focused and the row is collapsed.
```suggestion
			const cannotFocusNextColumn =
			        canExpandCollapse &&
				activeRow.getAttribute( 'aria-expanded' ) === 'false' &&
				keyCode === RIGHT;
```

The reason being that if a TreeGrid has three columns, the code here would prevent moving from the second to third columns on collapsed rows.

Reply:
There is an example of a treegrid with three columns in the storybook if it helps to have a working example, though it doesn't have the collapse/expand feature implemented unfortunately so we can't reproduce this specific scenario:
https://wordpress.github.io/gutenberg/?path=/story/components-treegrid--default

Maybe I can add that later in the week so it's a better example.

Reply:
I wasn't even aware Gutenberg used Storybook. Interesting. Yes, that would be helpful to have as then we could have better predicted regressions.